scripts/squad-config

```diff
@@ -0,0 +1,502 @@
+    lookup_fields = ListProperty(default=['name'], exclude_if_none=True)
+
+
+class EmailTemplateConfig(JsonObject):
+    name = StringProperty()
+    subject = StringProperty()
+    plain_text = StringProperty()
+    html = StringProperty()
+
+    # meta fields, not to be pushed to SQUAD
+    lookup_fields = ListProperty(default=['name'], exclude_if_none=True)
+
+
+class UserGroupConfig(JsonObject):
+    name = StringProperty()
+
+    # meta fields, not to be pushed to SQUAD
+    lookup_fields = ListProperty(default=['name'], exclude_if_none=True)
+
+
+class EnvironmentConfig(JsonObject):
+    slug = StringProperty()
+    name = StringProperty()
+    expected_test_runs = IntegerProperty()
+    description = StringProperty()
+
+    # meta fields, not to be pushed to SQUAD
+    lookup_fields = ListProperty(default=['slug'], exclude_if_none=True)
+
+
+class ProjectConfig(JsonObject):
+    slug = StringProperty()
+    name = StringProperty()
+    is_public = BooleanProperty()
+    html_mail = BooleanProperty()
+    moderate_notifications = BooleanProperty()
+    custom_email_template = DictProperty()  # has to be filled after object creation
+    description = StringProperty()
+    important_metadata_keys = StringProperty()
+    enabled_plugins_list = StringProperty()
+    wait_before_notification = IntegerProperty()
+    notification_timeout = IntegerProperty()
+    notification_strategy = StringProperty(choices=['all', 'change'])
+
+    # related fields
+    environments = ListProperty(EnvironmentConfig)
+
+    # meta fields, not to be pushed to SQUAD
+    lookup_fields = ListProperty(default=['slug'], exclude_if_none=True)
+    # contains dict in form
+    # field_name: endpoint_name
+    related_fields = DictProperty(
+        default={'custom_email_template': 'emailtemplates'},
+        exclude_if_none=True
+    )
+
+
+class GroupConfig(JsonObject):
+    slug = StringProperty()
+    name = StringProperty()
+    description = StringProperty()
+
+    # related fields
+    projects = ListProperty(ProjectConfig)
+
+    # meta fields, not to be pushed to SQUAD
+    lookup_fields = ListProperty(default=['slug'], exclude_if_none=True)
+
+
+class SquadConfig(JsonObject):
+    backends = ListProperty(BackendConfig)
+    emailtemplates = ListProperty(EmailTemplateConfig)
+    usergroups = ListProperty(UserGroupConfig)
+    groups = ListProperty(GroupConfig)
+
+
+class SquadConfigException(Exception):
+    pass
+
+
+class ConfigWorker(object):
+    def __init__(self, args):
+        self.url = args.url
+        urlparts = urlsplit(self.url)
+        self.base_url = urlparts.netloc
+        self.url_scheme = urlparts.scheme
+
+        connection_token = "Token %s" % self.__get_connection_token(args.url)
+        self.headers = {
+            "Authorization": connection_token
+        }
+        self.old_config = None  # SquadConfig()
+        self.old_backend_config = {}
+
+    def __get_connection_token(self, url):
```

Comment:
the usual convention for private methods is to also finish the method name with two underscores (`__`)

(ditto for the several other private methods)

Reply:
ACK